analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Throw.cs

```diff
@@ -0,0 +1,785 @@
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInTry"";
+}
+finally
+{
+    tag = ""InFinally"";
+}
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InFinally",                // With Exception thrown by Tag("InTry")
+                "InFinally");               // With Exception thrown by `throw`
+
+            ValidateHasOnlyUnknownExceptionAndSystemException(validator, "InFinally");
+        }
+
+        [TestMethod]
+        public void Finally_ThrowInFinally()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+}
+finally
+{
+    tag = ""InFinally"";
+    throw new System.Exception();
+    tag  = ""UnreachableInFinally"";
+}
+tag = ""UnreachableAfterFinally"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InFinally");
+
+            validator.TagStates("InFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInTry"";
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch",          // With Exception thrown by `throw`
+                "AfterCatch");
+
+            ValidateHasOnlyUnknownExceptionAndSystemException(validator, "InCatch");
+            validator.TagStates("AfterCatch").Should().HaveCount(1).And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock_ReThrow()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw new System.Exception();
+    tag = ""UnreachableInTry"";
+}
+catch
+{
+    tag = ""InCatch"";
+    throw;
+}
+tag = ""UnreachableAfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch");         // With Exception thrown by `throw`
+
+            validator.ValidateExitReachCount(1);
+
+            validator.TagStates("InCatch").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasSystemException(x));
+        }
+        [TestMethod]
```

Comment:
An empy line was gone missing during the move 😆 
```suggestion
        }
        
        [TestMethod]
```